sdk/src/ed25519_instruction.rs

```diff
@@ -0,0 +1,287 @@
+#![cfg(feature = "full")]
+
+use crate::instruction::Instruction;
+use ed25519_dalek::{ed25519::signature::Signature, Signer, Verifier};
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Debug, PartialEq)]
+pub enum Ed25519Error {
```

Comment:
`thiserror` is helpful and what we've using on recently added error types

Reply:
Renamed `Ed25519Error` to `ThisError`